polaris.shopify.com/src/pages/components/index.tsx

```diff
@@ -1,8 +1,30 @@
+import type { ReactElement, ReactNode } from 'react'
+
 import type { NextPage } from "next";
 import ComponentsPage from "../../components/ComponentsPage";
+import LeftNavLayout from '../../components/LeftNavLayout'
+import {
+  getComponentNav,
+} from "../../utils/various";
+import { NavItem } from "../../components/Nav/Nav";
 
 interface Props {}
 
-const Components: NextPage<Props> = () => <ComponentsPage />;
+const navItems: NavItem[] = getComponentNav();
+
+type NextPageWithLayout = NextPage & {
+  getLayout?: (page: ReactElement) => ReactNode
+}
+
+const Components: NextPageWithLayout = () => <ComponentsPage />;
```

Comment:
Someone that know typescript better than me pls double check that this makes any sense at all 🙏